flips/20220811-fungible-tokens-metadata.md

```diff
@@ -0,0 +1,210 @@
+                providerLinkedType.isSubtype(of: Type<&{FungibleToken.Provider, Resolver}>()): "Provider type must include NonFungibleToken.Provider and FungibleMetadataViews.Resolver interface."
+            }
+            self.storagePath=storagePath
+            self.publicPath=publicPath
+            self.providerPath = providerPath
+            self.publicLinkedType=publicLinkedType
+            self.providerLinkedType = providerLinkedType
+            self.createEmptyVault = createEmptyVaultFunction
+        }
+    }
+```
+
+In order to be able to use the `FTDisplay` we will need to re-use or re-implement (depending on if this ends up being part of the original `MetadataViews.cdc` contract or having its own contract) the following views: `ExternalURL`, `File`, `HTTPFile`, `IPFSFile`, `Media` and `Medias`. 
+
+### Drawbacks
+
+Main drawback could came from inefficient implementations of views from fungible token contracts.
+
+### Alternatives Considered
+
+Since the `MetadataViews` contract has been a successful solution for increasing NFT interoperability no alternatives have been considered for the FT. 
+
+### Performance Implications
+
+Due to the potentially dynamic nature of this proposal, the methods that return metadata views can be implemented in any way that the author sees fit which could lead to some non-performant implementations.
+
+### Dependencies
+
+This proposal builds on the existing FT interface defined [here](https://github.com/onflow/flow-ft).
+
+### Engineering Impact
+
+Adhering to this standard will require contract authors to implement a few new methods as part of their FT. They will also likely need to import standard types into their contracts.
+
+There may also need to be a committee, process, and code repository put in place for managing the standard metadata types.
+
+### Best Practices
+
+Fungible Token issuers should implement the `FTView` to achieve the as much ecosystem compatibility as possible.
+
+### Tutorials and Examples
+
+Guidelines on how to implement the new views will be added to the `ExampleToken.cdc` contract, and their used will be documented on the `flow-ft` repo docs.
+
+### Compatibility
+
+Fungible Token contract owners will need to update them to add implementations for the new methods.
+
+### User Impact
+
+This feature can be rolled out with no fear of changes to the user. However documentation should be provided on how they can change their contracts and add views.
+
+## Related Issues
+
+
+
+## Prior Art
+
+The already working NFT Metadata views [standard](https://github.com/onflow/flow/blob/master/flips/20210916-nft-metadata.md) has been used as the main source for this proposal.
+
+## Questions and Discussion Topics
+
+Main friction point will be if we want this standard to be in a new different contract of if we want it to be part of the existent `MetadataViews.cdc` contract.
```

Comment:
I think the new views should be a new contract, but like I said above, reuse some of the primitive views.

Reply:
Should we discuss this on the forum or should we start the discussion assuming that what makes more sense is to put the new views in a different contract rather than updating the NFT one?

Reply:
We should definitely put the FT specific views in a new contract

Reply:
so maybe we should remove that from the FLIP to avoid any confusion?

Reply:
No, they should definitely stay in this FLIP because this is the FLIP that proposes them. I just mean that when we're writing the code, they should be in their own contract in the flow-ft repo instead of in the `MetadataViews` contract in the flow-nft repo

Reply:
oh yeah. sorry about the misunderstanding. We can remove that line